File: Simulation/Bronze/P2_2016-Circular_Barn/cbarn.py
with open("cbarn.in", "r") as fin:
	# Read in total number of rooms (N)
	N = int(fin.readline())
	# Read in the number of cows in each room
	rooms = [int(fin.readline()) for _ in range(N)]

# Try each room as the first room to unlock
min_dist = float("inf")
for unlock in range(N):
	dist = 0
	for r in range(N):
		dist += rooms[(unlock + r) % N] * r
	min_dist = min(min_dist, dist)

print(min_dist, file=open("cbarn.out", "w"))

File: Simulation/Bronze/P2_2016-Circular_Barn/cbarn.cpp
#include <bits/stdc++.h>
#define MAX_ROOMS 1001

using namespace std;

int main() {
	freopen("cbarn.in", "r", stdin);
	// Read in total number of rooms
	int n;
	cin >> n;
	// Read in the number of cows in each room
	int rooms[MAX_ROOMS];
	for (int i = 0; i < n; i++)
		cin >> rooms[i];

	// Try each room as the first room to unlock
	int minDist = INT_MAX;
	for (int unlock = 0; unlock < n; unlock++) {
		int dist = 0;
		for (int r = 0; r < n; r++) 
			dist += rooms[(unlock + r) % n] * r;
		minDist = min(minDist, dist);
	}

	freopen("cbarn.out", "w", stdout);
	cout << minDist << endl;
	return 0;
}
